### ISSUE TITLE
Throw error when logical insertions violate schema

### ISSUE THREAD

**Issue by alex-dixon** [CoNarrative/precept#91]
- Throw error when attempting to override facts as the result of a logical insertion
From our fullstack example:
```clj
; cart total when no percent total discount
(define [:app :cart/total ?total] :- [#_:not [_ :active-discount]]
  [[_ :summed-subtotals ?total]])

; cart total when percent total discount
(define [:app :cart/total (total-with-percent-off ?total ?percent-discount)]
  :- [[_ :active-discount ?id]]
  [[?id :discount/percent-total ?percent-discount]]
  [[_ :summed-subtotals ?total]])
```
The above throws error with the following message:
> Conflicting logical fact. You may have rules whose conditions are not
        mutually exclusive that insert! the same e-a consequence. The conditions for logically
        inserting an e-a pair must be exclusive if the attribute is one-to-one. If you have two
        identical accumulators and you are seeing this error, create a separate rule that inserts a
        fact with the accumulator's result and replace the duplicate accumulators with expressions
        that match on that fact.

Accordingly, adding `:not` back makes the conditions mutually exclusive. The error will not display in this case as we will not be attempting to override an existing fact in one of our indexes with another fact that is the result of a logical insertion. We are now updating our indexes when Clara signals a logical retraction is taking place. This was a missing piece in maintaining synchronization previously which appeared to allow us to maintain eventual consistency. As a result of this addition our schema based truth maintenance via e - a - Tuple and a - v - Tuple indices should accurately represent Clara's state at all times.

Importantly, we allow overwrites of facts with `insert-unconditional!` or `insert` (they are fundamentally different semantically than insert logical / insert!). We will further guard against logical errors by restricting unconditional insertions to overwrite e - a pairs if they were inserted as a result of unconditional insertions only. At present, they can overwrite a fact inserted via insert logical.

tl;dr 
- insert! / insert logical cannot overwrite an existing e - a pair (error will be thrown)
- insert / insert-unconditional! can overwrite anything, but we will soon be limiting this to overwriting only facts that were inserted unconditionally (i.e. anything but logical)

- Found another case in which logically inserted facts were conflicting. Clara appears to allow node sharing for accumulators. Having two identical accumulators in a condition of a rule that resulted in a logical insertion would produce "conflicting" insert logical facts. Like in the case above, this makes it possible to end up with more than one to one fact in the session. Fixed `entities` macro to not reuse the same accumulator from the rule that it rewrites. Added error message suggesting not using two identical accumulators in different rules.
